### PR TITLE
feat(costModels): add the cost models details page

### LINKED ISSUE #881: Price list page
## User Story
As user, I want to see my rates defined via a price list page.

## Impacts
UI, Docs

## Role
User

## Assumptions
- Koku API provides support for listing current rates

## UI Details
Each price list name is a link to the price list details page.

See mocks
https://redhat.invisionapp.com/share/NJS0RS3Q7KE#/screens/365184848

Screen Shot 2019-05-28 at 9 43 10 PM

## Acceptance Criteria (post summit)
- [ ] Link provided from Cost Management Settings
- [ ] The ability to filter rates
- [ ] The ability to paginate thru rates
- [ ] The ability to sort names, etc.

### PR DESCRIPTION
**Pull request by boaz0**
closes #881

### CODE REVIEW

**Comment by boaz0** [comment]
//ping @dlabrecq can you take a look? :pray:

**Comment by dlabrecq** [comment]
Silly question; how do I launch the cost modal modal? I'm looking for a link in the Insights navigation pane, like this:

Screen Shot 2019-08-05 at 9 37 17 AM

**Comment by dlabrecq** [comment]
Also want to note that when I click on the "View all price list" option here:

Screen Shot 2019-08-05 at 9 37 17 AM

The modal skeletons continuously load. In the browser console, I see 404 errors. It appears that the code continuously attempts to fetch upon error.

Screen Shot 2019-08-05 at 9 35 09 AM

**Comment by boaz0** [comment]
Thanks @dlabrecq for the review.
I opened an [issue](https://github.com/project-koku/koku-ui/issues/938) regarding `view price list` in `cloud details`.

About not seeing the `Cost models details` nav link, this is because it is right now deployed in the [CI beta](https://ci.cloud.paas.psi.redhat.com/beta/hybrid/cost-management/cost-models) and it will take time till it's merged.

According to Asa Price [here](https://github.com/RedHatInsights/cloud-services-config/pull/22#pullrequestreview-268438660):
> [...] that this will only affect the sidenav in the ci-beta environment until **the dynamic chrome nav gets merged up the chain** :)

File: src/pages/costModelsDetails/costModelsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          title: t('cost_models_details.table.actions.view'),
+          onClick: (event, rowId, rowData, extra) => null,
+        },
+        {
+          title: t('cost_models_details.table.actions.clone'),
+          onClick: (event, rowId, rowData, extra) => null,
+        },
+        {
+          title: t('cost_models_details.table.actions.delete'),
+          onClick: (event, rowId, rowData, extra) => null,
```
[the remark this answers is not in the document]
Comment (boaz0) [reply src/pages/costModelsDetails/costModelsTable.tsx]: I should remove this. I put that as a placeholder but till the delete/clone modals and the view cost model page aren't ready it should be deleted.

I tried to remove the `onClick` attribute but it seems as a requirement and causing the build to break.

**Comment by dlabrecq** [comment]
Ok, but how do I test the page? I also tried https://ci.foo.redhat.com:1337/beta/hybrid/cost-management/cost-models

**Comment by boaz0** [comment]
Hi @dlabrecq 

To make this work -- run `BRANCH=prod-beta yarn start` and visit https://ci.foo.redhat.com:1337/beta/hybrid/cost-management/cost-models 

You should see the Cost model details link in the nav sidebar.

**Comment by boaz0** [comment]
Hi @chambridge 
Can you take a look at this PR?

**Comment by dlabrecq** [comment]
Sorry, I was on PTO. Today, I was able to run the UI (per your instructions above) and view the cost models page.

1. I'm curious about the toolbar design. The cost model mocks (https://marvelapp.com/73ii7a7/screen/57709826) show a filter icon inside the "Name" button. The icon looks separate -- it's outside the dropdown?

2. Is there a PatternFly 4 example of a toolbar with a filter? We have so many variations, It would be good to know which one we should be using. 

3. I noticed that the filter looks out of sync with the Cost management sources page. For example, there is no filter icon there. And we have both primary ("Create a cost model") and secondary ("Add source") buttons. I'm curious why these pages are so different? Perhaps that is a question for @nlcwong ?

Screen Shot 2019-08-13 at 9 32 26 PM

4. The "View Details" and "Clone" options in the kebab don't appear to do anything, should it? I didn't dare test "Delete".

Screen Shot 2019-08-13 at 9 38 49 PM

4. The "Create a cost model" button does not open a PatternFly wizard as shown in the mocks. I'm curious if that functionality is still in the works?

**Comment by boaz0** [comment]
@dlabrecq you need to update your branch because I updated the PR.

* The wizard is in a different PR: https://github.com/project-koku/koku-ui/pull/944 and because of that the "create a cost model" button is supposed to be disabled now
* The kebab option was removed and will be added after the wizard will be merged since the delete and clone are using the same components the wizard has

**Comment by dlabrecq** [comment]
I spoke to @nlcwong regarding the filter icon and it looks a bit off. The design should match:

https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/filters#attribute-value-textbox-filters

Can that be fixed?

We also need to sync the Cost management sources toolbar -- I spoke to @nlcwong about creating an issue for that.

**Pull request merged**